crates/subspace-primitives/Cargo.toml

```diff
@@ -0,0 +1,27 @@
+[package]
```

Comment:
Why is this crate necessary, can't we import necessary types from the runtime instead? Also `subspace-primitives` is too generic, `subspace-runtime-primitives` would be more appropriate just like description says.

Reply:
Because `subspace-runtime` is pretty heavy and I don't want to pull in the whole runtime when I only need the primitives in some libraries. What's more, it's consistent with the upstream, I advocate for that. I can change it to `subspace-runtime-primitives` if you insist, but I have reservations about it.

Reply:
Makes sense. `subspace-runtime-primitives` would be nice because we already have other primitives, so having generic `subspace-primitives` becomes ambiguous.

Reply:
`polkadot-core-primitives` is conceptually a subset of `polkadot-primitives` and is re-exported in `polkadot-primitives`, it's kind of convenient to work with IMO.  What do you think if we also make `subspace-primitives` the top superset of all primitives and the downstream library will basically pull in `subspace-primitives` only? I don't have a strong opinion.

Reply:
Polkadot is different in that they only have a Substrate node to worry about.

We also have farmer and may have something else. So let's not mix things because of that. This is why `subspace-runtime-primitives` makes more sense to me and I don't see the point of bringing `subspace-core-primitives` into if it is currently perfectly fine without it.

Reply:
Updated, let's use `subspace-runtime-primitives` then.